src/pages/index.js

```diff
@@ -21,79 +21,188 @@ import {
   addCardModal,
   addCardFormElement,
   initialCards,
+  editAvatarButton,
+  avatarEditForm,
 } from "../utils/constants.js";
+import Api from "../components/Api.js";
+import PopupConfirmation from "../components/PopupConfirmation.js";
 
 const editFormValidator = new FormValidator(config, profileEditForm);
 const addFormValidator = new FormValidator(config, addCardFormElement);
+const avatarFormValidator = new FormValidator(config, avatarEditForm);
 
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
+avatarFormValidator.enableValidation();
 
-function createCard(item) {
-  const card = new Card(item, "#card-template", handleImageClick);
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "05177711-40d5-4e6a-abe7-ba1645f6efc9",
+    "Content-Type": "application/json",
+  },
+});
+
+function createCard(data) {
+  const card = new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCard,
+    handleLikeClick
+  );
   return card.getView();
 }
 
-const cardListEl = new Section(
-  {
-    items: initialCards,
-    renderer: (data) => {
-      cardListEl.addItem(createCard(data));
-    },
-  },
+function renderItems(items) {
+  items.forEach((item) => {
+    this._renderer(item);
+  });
+}
+let cardSection;
+api
+  .getInitialCards()
+  .then((cards) => {
+    // Initialize cardSection before using it
+    cardSection = new Section(
+      {
+        items: cards,
+        renderer: (item) => {
+          const card = createCard(item);
+          cardSection.addItem(card);
+        },
+      },
+      ".cards__list"
+    );
 
-  ".cards__list"
-);
+    cardSection.renderItems();
+  })
+  .catch(console.error);
 
-cardListEl.renderItems();
+const addCardPopup = new PopupWithForm("#add-card-modal", async (formData) => {
+  const name = formData.title;
+  const link = formData.Url;
+
+  const res = await api.createCard({ name, link });
+  cardSection.addItem(createCard(res)); // Use here after initialization
+
+  addNewCardButton.addEventListener("click", () => {
+    addFormValidator._disableSubmitButton();
```

Comment:
`_disableSubmitButton`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.